src/main/java/com/minecolonies/coremod/colony/buildings/BuildingBuilder.java

```diff
@@ -178,25 +179,26 @@ public void serializeToView(@NotNull ByteBuf buf)
      *
      * @return a new Hashmap.
      */
-    public Map<Block, Integer> getNeededResources()
+    public Map<IBlockState, Integer> getNeededResources()
     {
         return new HashMap<>(neededResources);
     }
 
     /**
      * Add a new resource to the needed list.
-     *
-     * @param res    the resource.
+     *  @param res    the resource.
      * @param amount the amount.
+     * @param metaData the metaData.
      */
-    public void addNeededResource(Block res, int amount)
+    public void addNeededResource(@Nullable final Block res, final int amount, final int metaData)
     {
         int preAmount = 0;
         if (this.neededResources.containsKey(res))
         {
             preAmount = this.neededResources.get(res);
         }
-        this.neededResources.put(res, preAmount + amount);
+        IBlockState state = res.getStateFromMeta(metaData);
```

Comment:
![MINOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png) Remove this use of "getStateFromMeta"; it is deprecated. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://home.kk-sc.de:9000/coding_rules#rule_key=squid%3ACallToDeprecatedMethod)
![MINOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png) Local variable 'state' could be declared final [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://home.kk-sc.de:9000/coding_rules#rule_key=pmd%3ALocalVariableCouldBeFinal)